pkg/network/public/public.go

```diff
@@ -349,6 +355,24 @@ func EnsurePublicSetup(nodeID pkg.Identifier, inf *pkg.PublicConfig) (*netlink.B
 	return br, netlink.LinkSetUp(br)
 }
 
+func ensureTestNamespace(publicBrdige *netlink.Bridge) error {
+	netNS, err := namespace.GetByName(testNamespace)
+	if err != nil {
```

Comment:
if you check `GetByName` implementation you will see that it returns os.ErrNotExist if the namespace does not exist. So that's the only valid case you should try to create the namespace, then any other error type should be returned

```go
netNS, err := GetByName(name)
if os.IsNotExist(err) {
   err = Create(name)
}

if err != nil {
   return err
}
```